includes/TingSearchCqlDoctor.class.inc

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/**
+ * @file
+ *
+ * File holds class to convert not valid searchphrases to valid cql.
+ * Private membet cql_string holds an internal reprensation of the original search string. All parts of the string
+ * is examined to check for healthy cql.
+ *
+ * Enqouted parts of the string stays the same. Parantheses are enqouted if content is not healthy. Blanks are
+ * replaced by 'and'. Mulitple whitespaces are removed. Reserved characters are enqouted.
+ *
+ * Examples not valid:
+ * portland (film) => portland and (film)
+ * henning mortensen (f. 1939) => henning and mortensen and "(f. 1939)"
+ * harry and (White night) = harry and "(White night)"
+ *
+ *
+ * Valid cql stays the same eg.
+ * dkcclterm.sf=v and dkcclterm.uu=nt and (term.type=bog) not term.literaryForm=fiktion
+ */
+
+/**
+ * Class TingSearchCqlDoctor.
+ *
+ * An attempt to convert non cql search phrases to valid cql.
+ */
+class TingSearchCqlDoctor {
+
+  /**
+   * @var string cql_string.
+   *   Internael representation of the searchphtrase.
+   */
+  private $cql_string;
+  /**
+   * @var array pattern.
+   *   The pattern to replace in searchstring.
+   */
+  private $pattern;
+  /**
+   * @var array replace.
+   *   The string to replace with.
+   */
+  private $replace;
+  /**
+   * @var int replace_key.
+   *   Used to prepend key - @see $this->get_replace_key().
+   */
+  private static $replace_key = 10;
+
+  /**
+   * Constructor,Escape reserved characters, remove multiple whitespaces
+   * and sets private member $cql_string with trimmed string.
+   *
+   * @param string $string
+   *  The search phrase to cure.
+   */
+  public function __construct($string) {
+    $this->cql_string = trim($string);
+    // handle reserved characters.
+    $this->escape_reserved_characters();
+    // Remove multiple whitespaces.
+    $this->cql_string = preg_replace('/\s+/', ' ', $this->cql_string);
+  }
+
+  /**
+   * Method to convert a string to strict cql.
+   *
+   * Basically this method adds quotes when needed.
+   *
+   * @param string $string
+   *   The search query.
+   * @return string
+   *   Cql compatible string.
+   */
+  public function string_to_cql() {
+    // hendle parantheses
+    $this->fix_paranthesis();
+    // handle qoutes
+    $this->fix_qoutes();
+    // format the string
+    $return = $this->format_cql_string();
+    return $return;
```

Comment:
If the $return variable is not evaluated or use, so you could simply return it.

<pre>
return $this->format_cql_string();
</pre>